src/cluster/master.go

```diff
@@ -24,39 +26,50 @@ func (c *Cluster) KingMe() {
 	c.Microcontrollers = append(c.Microcontrollers, me)
 	c.Me = c.Master()
 
-	//The Master waits ...
+	//The Master pulls out its stethoscope ...
+	go func() {
+		ticker := time.NewTicker(30 * time.Second)
+		for {
+			select {
+			case t := <-ticker.C:
+				log.Println("Begin Heartbeat Check", t)
+				for _, m := range c.Microcontrollers {
+					if !m.Master {
+						log.Println("Checking Peer:", m.Name, m.ToFullAddress())
+						url := "http://" + m.ToFullAddress() + "/v1/health"
+						resp, err := http.Get(url)
+						if err != nil || resp.StatusCode != 200 {
+							log.Println(m.Name + " @" + m.ToFullAddress() + " is NOT ok")
+							log.Println("Deregistering Microcontroller...")
+							c.RemoveMicrocontroller(m)
+							c.SendClusterUpdate([]mc.Config{})
+						} else {
+							log.Println(m.Name + " @" + m.ToFullAddress() + " is ok")
+						}
+					}
+				}
+			}
+		}
+	}()
 }
 
 //AddMicrocontroller attempts to add a microcontroller to the cluster and returns the response data. This should only be run by the master.
-func (c *Cluster) AddMicrocontroller(newMC mc.Config) (response PeerUpdateMessage, err error) {
+func (c *Cluster) AddMicrocontroller(newMC mc.Config) error {
```

Comment:
having the funcs that add and remove micros from the collection generate and send msgs felt wrong. Had to separate this work out

Reply:
Thats awesome. Totally agree